tests/simpleKVBC/TesterReplica/internalCommandsHandler.cpp

```diff
@@ -141,27 +159,27 @@ bool InternalCommandsHandler::executeWriteCommand(uint32_t requestSize,
   }
 
   SimpleKey *readSetArray = writeReq->readSetArray();
-  BlockId currBlock = m_storage->getLastBlock();
+  BlockId currBlock = m_storage->getLastBlockId();
 
   // Look for conflicts
   bool hasConflict = false;
   for (size_t i = 0; !hasConflict && i < writeReq->numOfKeysInReadSet; i++) {
-    m_storage->mayHaveConflictBetween(
-        buildSliverFromStaticBuf(readSetArray[i].key), writeReq->readVersion + 1, currBlock, hasConflict);
+    const auto key = std::string(readSetArray[i].key, KV_LEN);
+    const auto latest = m_storage->getLatestVersion(CAT_ID, key);
+    hasConflict = (latest && (latest->deleted || latest->version > writeReq->readVersion));
```

Comment:
Why don't we have to check the version for `latest->deleted`? What if the initial transaction read from a prior version that wasn't deleted?

Reply:
Actually, TesterReplica doesn't delete any keys and, AFAIK, this check is redundant anyway.

Reply:
I've changed that check to an assert as we never delete keys in TesterReplica at that stage.